Dockerfile

```diff
@@ -18,12 +18,14 @@ ENV GOPATH=/go
 WORKDIR /usr/src/signalfx-agent
 
 COPY go.mod go.sum ./
+COPY lib/go.mod lib/go.sum ./lib/
 RUN go mod download
 
 COPY cmd/ ./cmd/
 COPY scripts/collectd-template-to-go scripts/make-versions ./scripts/
 COPY Makefile .
 COPY pkg/ ./pkg/
+COPY lib/ ./lib/
```

Comment:
Convention is to put library code in pkg as described here: https://github.com/golang-standards/project-layout#pkg.  We put all the agent code in `pkg` because some people use it as a library.

Just put it in `pkg/core/correlation` or something.  You can still give it it's own go.mod if you really need it to be a separate module, just like we do in collector-contrib.

Reply:
I put it in pkg/apm (It already contains a package called correlation so easier to put it under something else).